### PR TITLE
feat: add shared Markdown configuration  and renderString shortcode (resolves #158)

### LINKED ISSUE #158: Markdown filter causes issues in dependent projects
## Is your feature request related to a problem?

The Markdown filter seems to be causing console errors in projects. This is probably because it loads Eleventy's Markdown template engine in an unsupported way.

## Describe the solution you'd like

The filter should be deprecated, and instead the plugin should offer an enhanced Markdown configuration which can be used in dependant projects (along the lines of this: https://github.com/nhoizey/eleventy-plugin-pack11ty/blob/main/_11ty/markdown.js).

## Describe alternative solutions you've considered

Not applicable.

## Additional context or notes

- https://github.com/inclusive-design/idrc/issues/634

### Existing implementations

- https://github.com/fluid-project/fluidproject.org/blob/main/src/filters/markdown-filter.js
- https://github.com/fluid-project/floeproject.org/blob/71e3d84d55076db395f92c0820bd9a633cb33513/eleventy.config.js#L134-L140
- https://github.com/fluid-project/handbook.floeproject.org/blob/8a715386f3959b7154ff293675c6afcbb2056b67/.eleventy.js#L37-L48
- https://github.com/inclusive-design/idrc/blob/dev/src/filters/markdown.js
- https://github.com/fluid-project/infusion-docs/blob/a40b52c487a5f6bbae38dc106e4fee64c4aefe8e/eleventy.config.js#L22-L39
- https://github.com/inclusive-design/wecount.inclusivedesign.ca/blob/dev/src/filters/markdown.js

### PR DESCRIPTION
**Pull request by greatislander**
- Adds shared Markdown configuration which will be used by internal template engine and shortcode
- Updates shortcode to use its own instance of `markdown-it` (resolves #158)
- Adds `renderString` shortcode to allow a variable containing Markdown to be rendered using Eleventy's render plugin

### CODE REVIEW

**Review by jobara — COMMENTED** [review]
Looks good and I did a quick test with the IDRC site and things seem to work. 

About the footnotes. Can you review the output to ensure that it's sufficiently accessible? It doesn't look like there's much of a label on the link backs. I'm also not sure if we need to add a heading or something to the footnotes section at the bottom. Lastly is something like `[1]` sufficient as a label for the link to the footnote? It might be, do you have thoughts?

**Comment by greatislander** [comment]
@jobara Now I'm wondering if we should use a totally different solution ,something like this: https://github.com/KittyGiraudel/eleventy-plugin-footnotes

It seems much better for accessibility. See: https://kittygiraudel.com/2020/11/24/accessible-footnotes-and-a-bit-of-react/

File: package.json [hunk 1]
```diff
@@ -40,10 +40,12 @@
     "rimraf": "5.0.1"
   },
   "dependencies": {
+    "@11ty/eleventy-plugin-webc": "0.11.1",
     "browserslist": "4.21.9",
     "esbuild": "0.18.13",
     "html-minifier": "4.0.0",
     "lightningcss": "1.21.5",
+    "markdown-it-footnote": "3.0.3",
```
Comment (greatislander) [inline package.json:48]: Remove this

File: README.md [hunk 2]
```diff
@@ -304,6 +354,21 @@ Output:
...
+```nunjucks
+{% set markdownString = "# Page Title" %}
+{% renderString markdownString, "md" %}
+```
+
+Result:
+
+```html
+<h1>Page Title</h1>
+```
```
Comment (jobara) [inline README.md:370]: Might be helpful to have an example that uses the front matter data.

File: index.js [hunk 3]
```diff
@@ -61,17 +71,41 @@ module.exports = {
...
+        eleventyConfig.addPlugin(pluginWebc);
+
         /** Filters */
         eleventyConfig.addFilter("formatDate", formatDateFilter);
         eleventyConfig.addFilter("isoDate", isoDateFilter);
         eleventyConfig.addFilter("limit", limitFilter);
-        eleventyConfig.addFilter("markdown", markdownFilter);
+        eleventyConfig.addFilter("markdown", function (value) {
+            // eslint-disable-next-line no-console
+            console.warn("This filter will be removed in a future version of eleventy-plugin-fluid. Use the renderString shortcode instead.");
```
Comment (jobara) [inline index.js:84]: Probably worth adding the name of the filter in the message. Or does that already get logged along with this?

File: index.js [hunk 4]
```diff
@@ -61,17 +71,41 @@ module.exports = {
...
+            return md.render(value);
+        });
         eleventyConfig.addFilter("slug", () => {
             throw new Error("`slug` filter is no longer supported. Please use `slugify`.");
         });
         eleventyConfig.addFilter("split", splitFilter);
 
         /** Shortcodes */
+        eleventyConfig.addShortcode("renderString", async function (content, format) {
+            if (["html", "md", "webc", "11ty.js", "liquid", "njk", "hbs", "mustache", "ejs", "haml", "pug"].includes(format)) {
```
Comment (jobara) [inline index.js:102]: Can we move this list of content/template types to the configuration. We can have these as the default, but that would provide a way for the integrator to add another one if they add more plugins for example.

**Review by jobara — APPROVED** [review]

**Pull request merged**